primitives/manta/src/assets.rs

```diff
@@ -14,128 +14,200 @@
-        + Default
-        + TypeInfo
-        + From<MultiLocation>
-        + Into<Option<MultiLocation>>;
-
-    /// The Fungible ledger implementation of this trait
-    type FungibleLedger: FungibleLedger<C>;
+    /// Asset Registry
+    ///
+    /// The trait we use to register Assets and mint assets.
+    type AssetRegistry: AssetRegistry<
+        AssetId = Self::AssetId,
+        Balance = Self::Balance,
+        Metadata = Self::StorageMetadata,
+        Error = DispatchError,
+    >;
+
+    /// Fungible Ledger
+    type FungibleLedger: FungibleLedger<
+        AccountId = C::AccountId,
+        AssetId = Self::AssetId,
+        Balance = Self::Balance,
+    >;
 }
 
-/// The metadata of a Manta Asset
-#[derive(Clone, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
-pub struct AssetRegistrarMetadata {
+/// Asset Storage Metadata
+#[derive(Clone, Debug, Decode, Encode, Eq, Hash, Ord, PartialEq, PartialOrd, TypeInfo)]
+pub struct AssetStorageMetadata {
+    /// Asset Name
     pub name: Vec<u8>,
+
+    /// Asset Symbol
     pub symbol: Vec<u8>,
+
+    /// Number of Decimals
     pub decimals: u8,
-    pub evm_address: Option<H160>,
+
+    /// Frozen Flag
+    ///
+    /// Whether or not transfers of the asset are allowed.
     pub is_frozen: bool,
-    pub min_balance: Balance,
-    /// `is_sufficient`: Whether a non-zero balance of this asset is deposit of sufficient
-    /// value to account for the state bloat associated with its balance storage. If set to
-    /// `true`, then non-zero balances may be stored without a `consumer` reference (and thus
-    /// an ED in the Balances pallet or whatever else is used to control user-account state
-    /// growth).
-    /// For example, if is_sufficient set to `false`, a fresh account cannot receive XCM tokens.
+}
+
+impl<B> From<AssetRegistryMetadata<B>> for AssetStorageMetadata {
+    #[inline]
+    fn from(source: AssetRegistryMetadata<B>) -> Self {
+        source.metadata
+    }
+}
+
+/// Asset Registry Metadata
+#[derive(Clone, Debug, Decode, Encode, Eq, Hash, Ord, PartialEq, PartialOrd, TypeInfo)]
+pub struct AssetRegistryMetadata<B> {
+    /// Asset Storage Metadata
+    pub metadata: AssetStorageMetadata,
+
+    /// Optional EVM Address for the Asset
+    pub evm_address: Option<H160>,
+
+    /// Minimum Balance
+    pub min_balance: B,
+
+    /// Sufficiency Flag
+    ///
+    /// This flag should be set to `true` whenever a non-zero balance of this asset is deposit of
+    /// sufficient value to account for the state bloat associated with its balance storage. If set
+    /// to `true`, then non-zero balances may be stored without a `consumer` reference (and thus an
+    /// existential deposit in the balances pallet or whatever else is used to control user-account
+    /// state growth).
+    ///
+    /// If this flag is set to `false`, a fresh account cannot receive XCM tokens.
     pub is_sufficient: bool,
 }
 
+/*
 impl Default for AssetRegistrarMetadata {
```

Comment:
As we discussed we probably don't need this, but if we keeping it let's make it more appropriate.